pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -321,6 +322,7 @@ private static long getRandomInitialDelayInSeconds() {
       AutoRebalanceStrategy.class.getName();
   private static final int DEFAULT_LEAD_CONTROLLER_RESOURCE_REBALANCE_DELAY_MS = 300_000; // 5 minutes
   private static final String DEFAULT_DIM_TABLE_MAX_SIZE = "200M";
+  private static final int DEFAULT_CONTROLLER_NUM_THREAD_POOL = 500;
```

Comment:
This magic number might not always work. Suggest falling back to cached pool when it is not configured